contracts/swap/swap.go

```diff
@@ -88,20 +99,9 @@ func (s *Swap) ValidateCode(ctx context.Context, b bind.ContractBackend, address
 	return nil
 }
 
-// Deploy a Swap contract
-func Deploy(auth *bind.TransactOpts, backend bind.ContractBackend, owner common.Address, harddepositTimeout time.Duration) (addr common.Address, s *Swap, tx *types.Transaction, err error) {
-	s = New()
-	addr, tx, s.Instance, err = contract.DeploySimpleSwap(auth, backend, owner, big.NewInt(int64(harddepositTimeout.Seconds())))
-	return addr, s, tx, err
-}
-
-// ContractParams returns contract information
-func (s *Swap) ContractParams() *Params {
-	return &Params{
-		ContractCode: contract.SimpleSwapBin,
-		ContractAbi:  contract.SimpleSwapABI,
-	}
-}
+// WaitFunc is the default function to wait for transactions
+// We can overwrite this in tests so that we don't need to wait for mining
+var WaitFunc = waitForTx
```

Comment:
It could, it is an option, but then every function would need it as a parameter. We have used this exact pattern of using a global var in two other places that is why I decided to use it.

Reply:
That clears it up. Thanks!